### DISCUSSION TITLE
MVIM - Pasting text ending with newline inserts text on previous line

### DISCUSSION THREAD

**Discussion by eric-marshall** [macvim-dev/macvim#1458] [Q&A]
FYI - I'm running on **Sonoma 14.1** using **MacVim r178** and no **vim**-specific initialization files.

First, from any application, I copy text that ends with a newline
Second, I launch **mvim** and enter insert mode, type some characters, then enter Command-V

The copied text gets inserted on the line **above** the characters I just typed, and the cursor is placed at the beginning of the line containing the typed characters, and insert mode is still active.

Why isn't the text simply inserted **after** the cursor, as I would expect?

Using the **MacVim**-supplied **vim** program within the **Terminal** application, the text **is** appended to the current line, after the typed characters.

Thanks in advance.

**Comment by ychin** [comment]
Cmd-V is mapped to `"+P`, which pastes the text before the cursor. This is because it makes more sense to do so when pasting non-line-based texts, as using `p` (instead of `P`) to put the text after the cursor will look like it skipped a character. It just happens that for line-based texts using `p` is probably better than `P` but it's just one of those Vim oddities as the before/after thing frequently trip people up. You can always manually do `"+p` instead.
